EasyPost/Client.cs

```diff
@@ -48,18 +47,9 @@ public Client(ClientConfiguration clientConfiguration, HttpClient? customHttpCli
             ServicePointManager.SecurityProtocol |= Security.GetProtocol();
             Configuration = clientConfiguration ?? throw new ArgumentNullException(nameof(clientConfiguration));
 
-            try
-            {
-                Assembly assembly = typeof(Client).Assembly;
-                FileVersionInfo info = FileVersionInfo.GetVersionInfo(assembly.Location);
-                _libraryVersion = info.FileVersion ?? "Unknown";
-            }
-            catch (Exception)
-            {
-                _libraryVersion = "Unknown";
-            }
-
-            _dotNetVersion = Environment.Version.ToString();
+            _libraryVersion = RuntimeInfo.ApplicationInfo.ApplicationVersion;
+            _dotNetVersion = RuntimeInfo.ApplicationInfo.DotNetVersion;
+            _osDetails = $"OS/{RuntimeInfo.OperationSystemInfo.Name} OSVersion/{RuntimeInfo.OperationSystemInfo.Version} OSArch/{RuntimeInfo.OperationSystemInfo.Architecture}";
```

Comment:
suggestion: what I meant was inline on line 38. Having each individual piece be modular so we can easily adjust if needed. So assigning them to a var like the lib version and then building each delimited piece of that string one by one so the engineer looking at it knows exactly what's included without needing to go elsewhere.